strategy-offline-fallback/service-worker.js

```diff
@@ -0,0 +1,71 @@
+var CACHE = 'offline-fallback';
+
+// On install, cache the non available resource.
+self.addEventListener('install', function(evt) {
+  console.log('The service worker is being installed.');
+  evt.waitUntil(precache().then(function () {
+    // Skip waiting is necessary since we want the SW to be active ASAP
+    // instead of waiting for non controlled clients to be closed first.
+    return self.skipWaiting();
+  }));
+});
+
+self.addEventListener('activate', function (evt) {
+  // `self.clients.claim()` allows the service worker to start intercepting
+  // requests immediately. In addition to `self.skipWaiting()` is needed to
+  // allow serving fallbacks since the beginning.
+  evt.waitUntil(self.clients.claim());
+});
+
+// Use any strategy you want but check for errors to finally use the fallback.
+self.addEventListener('fetch', function(evt) {
+  console.log('The service worker is serving the asset.');
+  evt.respondWith(whateverStrategy(evt.request).catch(function () {
+    return useFallback();
+  }));
+});
+
+// Open a cache and use `addAll()` with an array of assets to add all of them
+// to the cache. Extend the installation until all the assets are saved.
+function precache() {
+  return caches.open(CACHE).then(function (cache) {
+    return cache.addAll([
+      './controlled.html'
+    ]);
+  });
+}
+
+// This is a simplified version of network then cache without timeout to
+// illustrate error control.
+function whateverStrategy(request) {
```

Comment:
this is probably not a very fortunate name, as it doesn't describe what is going on and it's also distracting me from focusing on the code to understand what it does and come up with a suggestion for a better name :-/